core/flamingock-test-support/src/main/java/io/flamingock/internal/core/builder/AuditEntryExpectation.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2025 Flamingock (https://www.flamingock.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.flamingock.internal.core.builder;
+
+import io.flamingock.api.annotations.Apply;
+import io.flamingock.internal.common.core.audit.AuditEntry;
+import io.flamingock.internal.common.core.audit.AuditTxType;
+
+import java.time.LocalDateTime;
+
+/**
+ * Builder pattern for creating expected audit entry values in tests.
+ * Relocated into flamingock-test-support module.
+ */
+public class AuditEntryExpectation {
+
+    private String expectedExecutionId;
+    private String expectedStageId;
+    private String expectedTaskId;
+    private String expectedAuthor;
+    private LocalDateTime expectedCreatedAt;
+    private AuditEntry.Status expectedState;
+    private AuditEntry.ExecutionType expectedType;
+    private String expectedClassName;
+    private String expectedMethodName;
+    private Object expectedMetadata;
+    private Long expectedExecutionMillis;
+    private String expectedExecutionHostname;
+    private String expectedErrorTrace;
+    private Boolean expectedSystemChange;
+    private AuditTxType expectedTxType;
+    private String expectedTargetSystemId;
+
+    // Time range for flexible timestamp verification
+    private LocalDateTime timestampAfter;
+    private LocalDateTime timestampBefore;
+
+    // Flags for optional field verification
+    private boolean shouldVerifyExecutionId = false;
+    private boolean shouldVerifyStageId = false;
+    private boolean shouldVerifyTimestamp = false;
+    private boolean shouldVerifyExecutionMillis = false;
+    private boolean shouldVerifyExecutionHostname = false;
+
+    AuditEntryExpectation() {}
+
+    public static AuditEntryExpectation auditEntry() {
+        return new AuditEntryExpectation();
+    }
```

Comment:
Remove this, the minimum we require is state(given by the static constructors(`APPLIED`, `FAILED`, etc)) and the id